[BUG]: Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in

### What happened?

How To fix?

### Versions

Octokit.js v6.0.1, Node v18.19.1

### Relevant log output

```shell
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in /home/container/node_modules/@octokit/core/package.json
    at new NodeError (node:internal/errors:405:5)
    at exportsNotFound (node:internal/modules/esm/resolve:366:10)
    at packageExportsResolve (node:internal/modules/esm/resolve:656:13)
    at resolveExports (node:internal/modules/cjs/loader:584:36)
    at Module._findPath (node:internal/modules/cjs/loader:658:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1120:27)
    at Module._load (node:internal/modules/cjs/loader:975:27)
    at Module.require (node:internal/modules/cjs/loader:1225:19)
    at require (node:internal/modules/helpers:177:18)
    at Object.<anonymous> (/home/container/index.js:31:21) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
Node.js v18.19.1
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

@wolfy1339  I'm getting the same error and others, setting to ESM or "node16" in tsconfig
With: 
```
"target": "es2016"
"module": "commonjs"
```
Results in:
>[ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined 

With:
```
"target": "es2022"
"module": "es2022"
```
Case package.json type = "module":
>ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts".

Without package.json type = "module":
>SyntaxError: Cannot use import statement outside a module and Cannot find module '@octokit/core'. 

With:
```
"target": "es2022"
"module": "Node16"
```
Results in:  
>The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("@octokit/core")' call instead. To convert this file to an ECMAScript module, change its file extension to '.mts', or add the field `"type": "module"` to 'C:/Users/{user}/project/package.json'.

My default tsconfig.json file:
```
{
  "compilerOptions": {
    "target": "ES2016",                                  /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */
    "module": "CommonJS",                                /* Specify what module code is generated. */
    "rootDir": "./",                                  /* Specify the root folder within your source files. */
    "outDir": "./dist",                                   /* Specify an output folder for all emitted files. */
    "esModuleInterop": true,                             /* Emit additional JavaScript to ease support for importing CommonJS modules. This enables 'allowSyntheticDefaultImports' for type compatibility. */
    "forceConsistentCasingInFileNames": true,            /* Ensure that casing is correct in imports. */    
    "strict": true,                                      /* Enable all strict type-checking options. */
    "skipLibCheck": true                                 /* Skip type checking all .d.ts files. */
  }
}
```
I'm using **"@octokit/core": "^6.0.1"**  and **Node v18.17.0** 
my import look like this:
```
import { Octokit } from "@octokit/core"
const octokit = new Octokit({ auth: env.GITHUB_TOKEN });
```

--- a/README.md
+++ b/README.md
@@ -55,6 +55,8 @@ import { Octokit } from "@octokit/core";
 </tbody>
 </table>
 
+As we use [conditional exports](https://nodejs.org/api/packages.html#conditional-exports), you will need to adapt your `tsconfig.json`. See the TypeScript docs on [package.json "exports"](https://www.typescriptlang.org/docs/handbook/modules/reference.html#packagejson-exports).
+
 ### REST API example
 
 ```js

--- a/scripts/build.mjs
+++ b/scripts/build.mjs
@@ -62,6 +62,9 @@ async function main() {
             import: "./dist-src/index.js",
             default: "./dist-src/index.js"
           },
+          "./types": {
+            types: "./dist-types/types.d.ts"
+          }
         },
         sideEffects: false,
       },

--- a/scripts/build.mjs
+++ b/scripts/build.mjs
@@ -55,12 +55,12 @@ async function main() {
       {
         ...pkg,
         files: ["dist-*/**", "bin/**"],
-        main: "./dist-src/index.js",
         types: "./dist-types/index.d.ts",
         exports: {
           ".": {
             types: "./dist-types/index.d.ts",
             import: "./dist-src/index.js",
+            default: "./dist-src/index.js"
           },
         },
         sideEffects: false,

--- a/src/index.ts
+++ b/src/index.ts
@@ -2,7 +2,7 @@ import { getUserAgent } from "universal-user-agent";
 import type { HookCollection } from "before-after-hook";
 import Hook from "before-after-hook";
 import { request } from "@octokit/request";
-import { graphql, withCustomRequest } from "@octokit/graphql";
+import { type graphql, withCustomRequest } from "@octokit/graphql";
 import { createTokenAuth } from "@octokit/auth-token";
 
 import type {

